tensorflow/python/keras/engine/training.py

```diff
@@ -2211,6 +2217,40 @@ def _updated_config(self):
     }
     return model_config
 
+  def __reduce_ex__(self, protocol):
+    """Support for Pythons's Pickle protocol.
+    """
+    save_folder = f"tmp/saving/{id(self)}"
+    ram_prefix = "ram://"
+    temp_ram_location = os.path.join(ram_prefix, save_folder)
+    self.save(temp_ram_location)
+    b = BytesIO()
+    with zipfile.ZipFile(b, "w", zipfile.ZIP_DEFLATED) as zf:
+        for _, _, filenames in tf_io.file_io.walk_v2(temp_ram_location):
+            for filename in filenames:
+                with gfile.GFile(filename, "rb") as f:
+                    zf.writestr(os.path.relpath(filename, temp_ram_location), f.read())
+    b.seek(0)
+    return self._reconstruct_pickle, (np.asarray(memoryview(b.read())), )
+
+  @classmethod
+  def _reconstruct_pickle(cls, obj):
+    """Reconstruct a model from the result of __reduce_ex__
+    """
+    save_folder = f"tmp/saving/{id(obj)}"
+    ram_prefix = "ram://"
+    temp_ram_location = os.path.join(ram_prefix, save_folder)
+    b = BytesIO()
+    b.write(obj)
+    b.seek(0)
```

Comment:
We could simplify this a bit by initializing `BytesIO` with the `bytes`.

```suggestion
    b = BytesIO(obj)
```